src/config/config.ts

```diff
@@ -0,0 +1,139 @@
+import parse from 'parse-duration';
+import Web3 from 'web3';
+
+export interface Config {
+  chainConfigs: aloeChainConfig[];
+  multicallAddress: Address;
+  createAccountTopicID: string;
+  initialDeploy: number;
+  pollingInterval: number;
+  processLiquidatableInterval: number;
+  heartbeatInterval: number;
+  heartbeatTimeout: number;
+  clientKeepAliveTimeout: number;
+  sanityCheckInterval: number;
+  reconnectDelay: number;
+  reconnectMaxAttemmpts: number;
+  errorThreshold: number;
+  restartTimeout: number;
+}
+
+interface aloeChainConfig {
+  chainName: string;
+  chainNumber: number;
+  factory: Address;
+  borrowerLens: Address;
+  borrower: Address;
+}
+
+type Address = string;
+type ConditionFunc<T> = (input: T) => boolean;
+
+const defaultTimeUnit = 'millisecond';
+const emptyString = '';
+
+const greaterThanZero = (input: number) => {
+  return input !== undefined && input > 0;
+};
+const greaterThanOrEqualToZero = (input: number) => {
+  return input !== undefined && input >= 0;
+};
+const isAddress = (input: string, chainNumber?: number) => {
+  return input !== emptyString && Web3.utils.isAddress(input, chainNumber);
+};
+const isNotEmpty = (input: string) => {
+  return input !== emptyString;
+};
+
+export function readConfig(json: any): Config | null {
+  try {
+    const config: Config = parseConfig(json);
+    if (isValidConfig(config)) {
+      return config;
+    }
+  } catch (error) {
+    console.error('Error parsing JSON to config:', error);
+  }
+  return null;
+}
+
+function parseConfig(obj: any): Config {
+  return {
+    chainConfigs: obj.chainConfigs.map((item: aloeChainConfig) => item),
+    multicallAddress: obj.multicallAddress,
+    createAccountTopicID: obj.createAccountTopicID,
+    initialDeploy: obj.initialDeploy,
+    pollingInterval: getDuration(obj.pollingInterval),
+    processLiquidatableInterval: getDuration(obj.processLiquidatableInterval),
+    heartbeatInterval: getDuration(obj.heartbeatInterval),
+    heartbeatTimeout: getDuration(obj.heartbeatTimeout),
+    clientKeepAliveTimeout: getDuration(obj.clientKeepAliveTimeout),
+    sanityCheckInterval: getDuration(obj.sanityCheckInterval),
+    reconnectDelay: getDuration(obj.reconnectDelay),
+    reconnectMaxAttemmpts: obj.reconnectMaxAttemmpts,
+    errorThreshold: getDuration(obj.errorThreshold),
+    restartTimeout: getDuration(obj.restartTimeout),
+  } as Config;
+}
+
+function getDuration(quantity: string): number {
+  const duration = parse(quantity, defaultTimeUnit);
+  if (!duration) {
+    return 0;
```

Comment:
Not certain that this function is needed. We might just be able to call `parse` directly on the fields.

Reply:
Follow up - if we return 0 then we might run into troubles with our `greaterThanOrEqualToZero` check. I think it might be better to just call `parse` directly with `defaultTimeUnit`.